scripts/globals/items/chocobo_shirt.lua

```diff
@@ -0,0 +1,42 @@
+-----------------------------------------
+--    ID: 10293
+--    Chocobo Shirt
+--    Dispense: Crystals
+-----------------------------------------
+
+-----------------------------------------
+-- OnItemCheck
+-----------------------------------------
+
+function onItemCheck(target)
+    local result = 0;
+    if (target:getFreeSlotsCount() == 0) then
+        result = 308;
+    end
+    return result;
+end;
+
+-----------------------------------------
+-- OnItemUse
+-----------------------------------------
+
+function onItemUse(target)
+local dayElement = VanadielDayElement();
+-- day order: 0. fire:4096, 1. earth:4099, 2. water:4101, 3. wind:4098, 4. ice:4097, 5. thunder:4100, 6. light:4102, 7. dark:4103
+if (dayElement == 0)
+    target:addItem(4096,math.random(4,12)); then
+elseif (dayElement == 1)
+    target:addItem(4099,math.random(4,12)); then
+elseif (dayElement == 2)
+    target:addItem(4101,math.random(4,12)); then
+elseif (dayElement == 3)
+    target:addItem(4098,math.random(4,12)); then
+elseif (dayElement == 4)
+    target:addItem(4097,math.random(4,12)); then
+elseif (dayElement == 5)
+    target:addItem(4100,math.random(4,12)); then
+elseif (dayElement == 6)
+    target:addItem(4102,math.random(4,12)); then
+else (dayElement == 7)
+    target:addItem(4103,math.random(4,12)); then
```

Comment:
Thanks for the advice. It looks much better now. But, one question, why the VanadielDayElement()**+1**? Also, the array should start at 0 for fireday, correct? I tested my if statement, and it worked, so the material I found was correct. Fireday = day 0, ++.